resources/js/features/game-list/components/DataTablePagination/DataTablePagination.tsx

```diff
@@ -68,28 +74,82 @@ export default function DataTablePagination<TData>({
     const canPrefetchPrevious = prefetchDirections?.includes('previous') && newPageIndex > 0;
 
     if (canPrefetchNext) {
-      prefetchPagination(Math.min(newPageIndex + 1, lastPageIndex));
+      prefetchPagination({
+        newPageIndex: Math.min(newPageIndex + 1, lastPageIndex),
+        newPageSize: pagination.pageSize,
+      });
     }
     if (canPrefetchPrevious) {
-      prefetchPagination(Math.max(newPageIndex - 1, 0));
+      prefetchPagination({
+        newPageIndex: Math.max(newPageIndex - 1, 0),
+        newPageSize: pagination.pageSize,
+      });
     }
 
     scrollToTopOfPage();
   };
 
+  const handlePageSizeChange = (newPageSize: number) => {
+    // If the user is changing the page size and they're not on the first page,
+    // auto-scroll to the top. Otherwise, things can quickly get disorienting.
+    if (pagination.pageIndex !== 0) {
+      scrollToTopOfPage();
+    }
+
+    table.setPagination({ pageIndex: 0, pageSize: newPageSize });
+  };
+
   return (
-    <div className="flex items-center justify-between">
+    <div className="flex items-center justify-center sm:justify-between">
+      {/* TODO X of Y rows selected */}
       <div />
 
-      <div className="flex items-center gap-6 lg:gap-8">
+      <div className="flex flex-col items-center gap-2 sm:flex-row sm:gap-6 lg:gap-8">
+        <div className="flex items-center gap-2">
+          <label id="rows-per-page-label" htmlFor="rows-per-page-select">
+            {t('Rows per page')}
+          </label>
+
+          <BaseSelect
+            value={`${table.getState().pagination.pageSize}`}
+            onValueChange={(value) => {
+              handlePageSizeChange(Number(value));
+            }}
+          >
+            <BaseSelectTrigger
+              id="rows-per-page-select"
+              aria-labelledby="rows-per-page-label"
+              className="h-8 w-[70px]"
+            >
+              <BaseSelectValue placeholder={table.getState().pagination.pageSize} />
+            </BaseSelectTrigger>
+
+            <BaseSelectContent side="top">
+              {[25, 50, 150].map((pageSize) => (
```

Comment:
Would it be reasonable to have a smaller value (like 10) to eliminate scrolling entirely?

Reply:
Having a smaller value like 10 sounds like a good idea.

150 and not 100 is a fair question. In latest, I've updated the possible page sizes to:
```
private const VALID_PAGE_SIZES = [10, 25, 50, 100, 200];
```